pkg/diagnostics/analyzers.go

```diff
@@ -254,6 +255,40 @@ func (a *analyzerFactory) capiKubeadmControlPlaneSystemLogAnalyzers() []*Analyze
 	}
 }
 
+// EksaDiagnosticsLogAnalyzers currently can help to analyze whether user is using a valid control plane IP to connect
+// to API server
+func (a *analyzerFactory) eksaDiagnosticsLogAnalyzers() []*Analyze {
+	runPingPod := "run-ping"
+	runPingPodLog := fmt.Sprintf("%s.log", runPingPod)
+	fullRunPingPodLogPath := path.Join(runPingPod, runPingPodLog)
+	return []*Analyze{
+		{
+			TextAnalyze: &textAnalyze{
+				analyzeMeta: analyzeMeta{
+					CheckName: fmt.Sprintf("%s:  Destination Host Unreachable. Log: %s", logAnalysisAnalyzerPrefix, fullRunPingPodLogPath),
+				},
+				FileName:     fullRunPingPodLogPath,
+				RegexPattern: `exit code: 0`,
+				Outcomes: []*outcome{
+					{
+						Fail: &singleOutcome{
+							When:    "false",
+							Message: fmt.Sprintf("API server is unavailable or the route to API Server is invalid. See %s", fullRunPingPodLogPath),
```

Comment:
Let's be more specific and say the control plane endpoint host is unreachable instead.